File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
private:
    int n;
    string recursive(int i, string curr){
        // Base case
        if (i == n) return curr;

        string next = "";
        char currNum = '\0';
        int count = 0;

        for (const char& c : curr){
            if (currNum == '\0' || currNum == c){
                ++count;
            }
            else{
                next += to_string(count) + currNum;
                count = 1;
            }
            currNum = c;
        }

        next += to_string(count) + currNum;

        return recursive(i + 1, next);
    }

public:
    string countAndSay(int n) {
        this -> n = n;
        return recursive(1, "1");
    }
};

File: 0038-count-and-say/0038-count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        def recursive(i: int, curr: str) -> str:
            if i == n:
                return curr

            nxt, curr_num, cnt = "", '', 0
            for c in curr:
                if curr_num == "" or c == curr_num:
                    cnt += 1
                else:
                    nxt += str(cnt) + curr_num
                    cnt = 1
                curr_num = c
            
            nxt += str(cnt) + curr_num
            return recursive(i + 1, nxt)
        
        return recursive(1, '1')